valhalla/thor/dijkstras.h

```diff
@@ -0,0 +1,273 @@
+
+  /**
+   * Compute the best first graph traversal to a list of destination locations
+   * @param  origin_locs  List of origin locations.
+   * @param  graphreader  Graphreader
+   * @param  mode_costing List of costing objects
+   * @param  mode         Travel mode
+   */
+  void ComputeReverse(google::protobuf::RepeatedPtrField<valhalla::Location>& dest_locations,
+                      baldr::GraphReader& graphreader,
+                      const std::shared_ptr<sif::DynamicCost>* mode_costing,
+                      const sif::TravelMode mode);
+
+  /**
+   * Compute the best first graph traversal from a list of origin locations using multimodal
+   * @param  origin_locs  List of origin locations.
+   * @param  graphreader  Graphreader
+   * @param  mode_costing List of costing objects
+   * @param  mode         Travel mode
+   */
+  void ComputeMultiModal(google::protobuf::RepeatedPtrField<valhalla::Location>& origin_locations,
+                         baldr::GraphReader& graphreader,
+                         const std::shared_ptr<sif::DynamicCost>* mode_costing,
+                         const sif::TravelMode mode);
+
+protected:
+  // Virtual function called when expanding a node
+  //
+  // Children can implement this to customize behaviour
+  virtual void ExpandingNode(baldr::GraphReader& graphreader,
+                             const sif::EdgeLabel& pred,
+                             const ExpandingNodeMiscInfo& info){};
+
+  // A child-class can implement this to configure how the graph expands
+  virtual RouteCallbackRecommendedAction RouteCallbackDecideAction(baldr::GraphReader& graphreader,
+                                                                   const sif::EdgeLabel& pred,
+                                                                   const InfoRoutingType route_type) {
+    return RouteCallbackRecommendedAction::no_action;
+  };
+
+  bool has_date_time_;
+  int start_tz_index_;   // Timezone at the start of the Dijkstras
+  sif::TravelMode mode_; // Current travel mode
+  uint32_t access_mode_; // Access mode used by the costing method
+
+  // For multimodal
+  bool date_set_;
+  bool date_before_tile_;
+  uint32_t date_;
+  uint32_t dow_;
+  uint32_t day_;
+  uint32_t max_transfer_distance_;
+  std::string origin_date_time_;
+  uint32_t start_time_;
+  std::unordered_map<std::string, uint32_t> operators_;
+  std::unordered_set<uint32_t> processed_tiles_;
+
+  // Current costing mode
+  std::shared_ptr<sif::DynamicCost> costing_;
+
+  // Vector of edge labels (requires access by index).
+  std::vector<sif::BDEdgeLabel> bdedgelabels_;
+  std::vector<sif::MMEdgeLabel> mmedgelabels_;
+
+  // Adjacency list - approximate double bucket sort
+  std::shared_ptr<baldr::DoubleBucketQueue> adjacencylist_;
+
+  // Edge status. Mark edges that are in adjacency list or settled.
+  EdgeStatus edgestatus_;
+
+  /**
+   * Initialize prior to computing the Dijkstrass. Creates adjacency list,
```

Comment:
s/Dijkstrass/Dijkstras (shortest) path ?

Reply:
:+1:  It's less of shortest path and more of expanding the graph in all directions. I'll tweak to `Dijkstras algorithm` perhaps?